tests/integration/debug/components.py

```diff
@@ -0,0 +1,95 @@
+import io
+from typing import Tuple, Dict, Union
+
+import numpy as np
+
+from jina.executors.crafters import BaseCrafter
+from jina.executors.decorators import single
+from .helper import _crop_image, _move_channel_axis, _load_image
+
+
+class ImageReader(BaseCrafter):
+    """
+    Load image file and craft it into image matrix.
+
+    :class:`ImageReader` loads the image from the given file
+        path and save the `ndarray` of the image in the Document.
+
+    :param channel_axis: the axis id of the color channel.
+        The ``-1`` indicates the color channel info at the last axis
+    """
+
+    def __init__(self, channel_axis: int = -1, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.channel_axis = channel_axis
+
+    @single(slice_nargs=2)
+    def craft(self, buffer: bytes, uri: str, *args, **kwargs) -> Dict:
+        """
+        Read image file and craft it into image matrix.
+
+        Read the image from the given file path that specified in `buffer` and save the `ndarray` of the image in
+            the `blob` of the document.
+
+        :param buffer: the image in raw bytes
+        :param uri: the image file path
+
+        """
+        from PIL import Image
+
+        if buffer:
+            raw_img = Image.open(io.BytesIO(buffer))
+        elif uri:
+            raw_img = Image.open(uri)
+        else:
+            raise ValueError('no value found in "buffer" and "uri"')
+        raw_img = raw_img.convert('RGB')
+        img = np.array(raw_img).astype('float32')
+        if self.channel_axis != -1:
+            img = np.moveaxis(img, -1, self.channel_axis)
+        return dict(blob=img)
+
+
+class CenterImageCropper(BaseCrafter):
+    """
+    Crop the image with the center crop box.
+
+    The coordinate is the same coordinate-system in the
+        :py:mode:`PIL.Image`.
+
+    :param target_size: Desired output size. If size
+        is a sequence like (h, w), the output size will
+        be matched to this. If size is an int, the
+        output will have the same height and width as
+        the `target_size`.
+    :param channel_axis: Axis for channel
```

Comment:
Maybe it's better to specify it's the color channel?

Reply:
This is copy-pasted from the Hub and only used in the test. I don't think we should modify them here